kentcdodds/cypress-testing-library
# feat(commands): use waitForElement; add jest & cypress tests

@kentcdodds I apologize for this one huge commit but I wanted to make
it all real, fast (in a few hours).

- change commands to use `waitForElement` from `dom-testing-library`
(updated to latest)
- update `kcd-scripts` dependency to latest
- add jest tests for basic smoke check on the exports
- add cypress tests for functionality check of the commands
- add npm scripts that run the tests in both development and CI
environments (somewhat convoluted because cypress hasn't got enough
hooks to spin up the fixture server from JavaScript; I didn't want to
write a Node app so I used `serve` which needs to be killed after
cypress exits but regardless of its exit code, so I used `npm-run-all`
and `fkill-cli` and a combination of npm scripts with
parallel/sequential execution)
